ariadne/src/Ariadne/Wallet/Cardano/Kernel/DB/AcidState.hs

```diff
@@ -151,6 +152,7 @@ applyBlock (blocksByAccount,meta) = runUpdateNoErrors $ zoom dbHdWallets $
         (\prefBlock -> zoom hdAddressCheckpoints $
                            modify $ Spec.applyBlock (prefBlock,meta))
         blocksByAccount
+        Map.empty -- we are not expecting to create new accounts in applyBlock
```

Comment:
What does it mean?
```
-- | For each of the specified accounts, create them if they do not exist,
-- and apply the specified function.
createPrefiltered
```

AFAIU, the whole purpose of `createPrefiltered` is to create new accounts, isn't it?

Reply:
`createPrefiltered` is a complicated helper. It basically takes a `Map HdAccountId PrefilteredUtxo` and some auxiliary instructions on what to do with the accounts & addresses after ensuring that they are created. I guess the way it was brought into existence is that someone at IOHK noticed that there were two very similar chunks of code in `applyBlock` and `createHdWallet`, so they extracted that code into a separate function. As long as this function is not exposed outside of the module, it remains an implementation detail, which is OK IMO.

Reply:
> the `createPrefiltered` function does create new accounts (at least sometimes)

This is true.

> But you say that we are not expecting to create new accounts. If we are not expecting it, why do we call `createPrefiltered`?

Like I said, `createPrefiltered` is a complicated helper. It both ensures that all the necessary accounts and addresses exist, and applies updates to them. In `applyBlock`, the _updates_ part is important. In `createHdWallet`, _creation_ matters.

> After ensuring that they have even been created, not in this particular call?

Yes.